// lib-folder/index.js
/* eslint-disable react-native/no-inline-styles */
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */
import React, { Component } from 'react';
import {
  Text,
  StyleSheet,
  View,
  TouchableOpacity,
  Dimensions
} from 'react-native';
var width = Dimensions.get('window').width; //full width
var height = Dimensions.get('window').height; //full height
export default class Footer extends Component {
  render() {
    return (
      <View style={styles.root}>
        <View style={styles.footerTopView}>
          <View style={styles.circle} />
        </View>
        <View style={styles.footerBottomView}>
          <TouchableOpacity style={styles.buttonLayout}>
            <Text style={styles.buttonText}>Home</Text>
          </TouchableOpacity>
          <TouchableOpacity style={[styles.buttonLayout, { borderTopRightRadius: 30 }]}>
            <Text style={styles.buttonText}>Offer</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.buttonLayout}>

          </TouchableOpacity>
          <TouchableOpacity style={[styles.buttonLayout, { borderTopLeftRadius: 30 }]}>
            <Text style={styles.buttonText}>Cart</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.buttonLayout}>
            <Text style={styles.buttonText}>Profile</Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}


const styles = StyleSheet.create({
  root: {
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0
  },
  footerTopView: {
    width,
    height:
      height / 11
  },
  footerBottomView: {
    width,
    display: 'flex',
    flexDirection: 'row'
  },
  circle: {
    width: 80,
    height: 80,
    borderRadius: 80 / 2,
    alignSelf: 'center',
    backgroundColor: 'red',
    top: 28,
    borderWidth: 7,
    borderColor: 'white',
    zIndex: 1000,
    elevation: 1000,
    position: "absolute"
  },
  buttonLayout: {
    width: '20%',
    height: height / 11,
    backgroundColor: 'green',
    justifyContent: 'center'
  },
  buttonText: {
    textAlign: "center",
    color: '#ffffff',
    fontSize: 16,
    fontWeight: 'bold'
  }
});
